### ISSUE TITLE
Docker env support for MQTT Broker settings

### ISSUE THREAD

**Issue by jinfahua** [emqx/kuiper#271]
I am working on compose files that use MQTT Broker as the EdgeX Message Bus rather than ZMQ.
 
I have it all configured and working accept a way to tell Kuiper, from the compose file, that the Message Bus type is “mqtt”.
 
Do you have an environment variable to set the Message Bus type?
I tired EDGEX_TYPE=mqtt, but that didn’t seem to work.
 
I was able to shell into the container and modify the configuration YML file to prove it will work. Now just need a way to specify the type via environment variable in the compose file.

**Comment by zhanghongtong** [comment]
Hi, @jinfahua 
I need to know in which file `edgex_type = mqtt` needs to be placed

**Comment by jinfahua** [comment]
` type ` in `etc/sources/edgex.yaml`, and I think all of following configuration items can be set through Docker environment variable.

```
#  type: mqtt
#  optional:
#    ClientId: "client1"
#    Username: "user1"
#    Password: "password"
#    Qos: "1"
#    KeepAlive: "5000"
#    Retained: "true/false"
#    ConnectionPayload: ""
#    CertFile: ""
#    KeyFile: ""
#    CertPEMBlock: ""
#    KeyPEMBlock: ""
#    SkipCertVerify: "true/false"
```

**Comment by zhanghongtong** [comment]
Ok i get it, i will start to implement it later

**Comment by lenny-intel** [comment]
Yes, we will want to be able to set each of these. Thanks!

**Comment by zhanghongtong** [comment]
@jinfahua Because the shell is very inconvenient to operate the yaml file, so I need a Goalng script to get the environment variables and edit the yaml file
We can specify the format of the environment variable. If an environment variable matches this format, then modify the corresponding configuration items, as follows:

```
EDGEX__DEFAULT__TYPE    <---->    default.type in etc/sources/edgex.yaml

MQTT_SOURCE__DEMO_CONF__QOS    <---->    demo_conf.qos in etc/mqtt_source.yaml
```
Because there is a key with "_" in the configuration, two "__" are used as the hierarchical segmentation

or
```
EDGEX.DEFAULT.TYPE    <---->    default.type in etc/sources/edgex.yaml

MQTT_SOURCE.DEMO_CONF.QOS    <---->    demo_conf.qos in etc/mqtt_source.yaml
```
Use "." As the hierarchical segmentation

I noticed that some of the keys in the configuration items are named in camel case. I did n’t think about how to deal with them. Do you have a better idea?

**Comment by zhanghongtong** [comment]
You can refer to the practice of EMQX: https://github.com/emqx/emqx-rel/tree/master/deploy/docker#configuration

**Comment by lenny-intel** [comment]
> I noticed that some of the keys in the configuration items are named in camel case. I did n’t think about how to deal with them. Do you have a better idea?

FYI, in EdgeX we convert all setting path/names to upper case before comparing them to the environment variable name. So `EDGEX__DEFAULT__TYPE` matches `Default.Type` and also matches `default.type`.

**Comment by zhanghongtong** [comment]
@lenny-intel  I think it's a good idea
cc @jinfahua

**Comment by jinfahua** [comment]
It's fixed in 0.4.1, and close the issue.
